anndata/_core/aligned_mapping.py

```diff
@@ -47,7 +49,9 @@ def _ipython_key_completions_(self) -> List[str]:
     def _validate_value(self, val: V, key: str) -> V:
         """Raises an error if value is invalid"""
         for i, axis in enumerate(self.axes):
-            if self.parent.shape[axis] != val.shape[i]:
+            if (not isinstance(val, AwkArray)) and (  # don't validate awk arrays
+                self.parent.shape[axis] != dim_len(val, i)
+            ):
```

Comment:
Suggesting to validate awkward array dimensions and only allow awkward arrays that are fixed length in the dimension(s) to be aligned. I mean, otherwise there's no point in having it in `obsm` or `X`, could just dump it in `uns` in that case. 

```suggestion
            if isinstance(val, AwkArray):
                try:
                    # can only align awkward arrays that are fixed length in the corresponding dimension(s)
                    val = ak.to_regular(val, i)
                except ValueError:
                    raise ValueError(f"Awkward array passed for key {key!r} is of variable length in dimension {i}")
            
            if self.parent.shape[axis] != ak.size(val, i):
             
```